odoo/models.py

```diff
@@ -4070,7 +4070,7 @@ def _check_company(self, fnames=None):
                 corecord = record.sudo()[name]
                 if corecord:
                     domain = corecord._check_company_domain(company)
-                    if domain and not corecord.with_context(active_test=False).filtered_domain(domain):
+                    if domain and corecord.with_context(active_test=False).filtered_domain(['!', *domain]):
```

Comment:
The complement of any domain is not working in all cases (yet).
```suggestion
                    if domain and corecord and not corecord.with_context(active_test=False).filtered_domain([domain]):
```

Reply:
Unfortunately, your suggested change would make the check incorrect again. Could you elaborate in which cases the `NOT` operator wouldn't work? The domain is quite simple (it's the result of calling `_check_company_domain`) so it might be fine after all.

Reply:
Sorry, since `corecord` should be renamed `corecords` because there may be multiple ones, that was why I made the mistake. You could compare the `len` of the records before and after filtering.

For information about what is still not complemented correctly even in 18.0, see #171291. In 17.0 from memory, comparisons and some field traversal are not correctly complemented. It may work in your case, it does not work in all cases and that would be hard to debug.

Reply:
I renamed `corecord` into `corecords` and adapted the check to avoid using domain complements (I used `!=` instead of `len` as it's slightly shorter). LMKWYT